docs/rfc-1-staking-contract.adoc

```diff
@@ -185,20 +195,56 @@ is higher than the stake amount on T staking contract.
 
 In case the stake has been slashed on the legacy contract, the operator is
 required to update their information on T staking contract as soon as possible.
-In practice, with the random beacon disabled, and tBTC v1 slashing the stake
-only in case of a proven fraud that had to be committed by all operators of
+In practice, for Keep, with the random beacon disabled, and tBTC v1 slashing the 
+stake only in case of a proven fraud that had to be committed by all operators of
 ECDSA keep, this approach is acceptable.
 
-An integral part of the staking contract should be a bot monitoring stakes and
-notifying about discrepancies. This is especially important given that the bot
-may need to voluntarily inform about discrepancies for operators that have been
-slashed to zero.
+An integral part of the staking contract should be a bot or process inside an
+off-chain client monitoring stakes and notifying about discrepancies. This is
+especially important given that the bot may need to voluntarily inform about
+discrepancies for operators that have been slashed to zero.
+
+Owner or operator can decrease the legacy contract active stake cached amount on
+T staking contract if no application has authorization higher than the liquid T
+stake. It allows to undelegate from the legacy staking contract while still
+being able to operate in T and earn rewards.
```

Comment:
What do you think about giving control on T, cached NU/KEEP separately in T staking contract? We can have some default rules when authorize T first, deauthorize NU/KEEP first and give ability to set this when needed?

Reply:
I do not see any obvious advantage of allowing it. With the existing top-up API and partial undelegations proposed in the [comment above](https://github.com/threshold-network/solidity-contracts/pull/15#discussion_r711199239), stakers have full control over their portfolio: what is the share of legacy stake and what is the share of T stake. They can partially withdraw or even eliminate the legacy stake from their portfolio entirely while still operating a node and earning rewards.

Given that - for simplicity - we always want to slash T stake first, allowing to split authorizations between legacy stake and T stake, would make the slashing quite complex.